MobiledgeX/Runtime/Scripts/MobiledgeXIntegration.cs

```diff
@@ -203,124 +172,123 @@ public async Task<bool> VerifyLocation()
-            if (findCloudletReply == null)
+            if(port == 0)
             {
-                Debug.Log("cannot find findCloudletReply");
+                port = appPortsDict.OrderBy(kvp => kvp.Key).First().Key;
             }
+            try {
 
-            Dictionary<int, AppPort> appPortsDict = me.GetTCPAppPorts(findCloudletReply);
-            int public_port = findCloudletReply.ports[0].public_port; // We happen to know it's the first one.
-            AppPort appPort = appPortsDict[public_port];
-            return await me.GetWebsocketConnection(findCloudletReply, appPort, public_port, 5000, path);
+                AppPort appPort = appPortsDict[port];
+                return await me.GetWebsocketConnection(findCloudletReply, appPort, port, 5000, path);
+            }
+            catch (KeyNotFoundException)
+            {
+                Debug.LogError("MobiledgeX: Port supplied is not mapped to your Application, Make sure the desired port is defined in your Application Port Mapping Section on MobiledgeX Console.");
+                throw new GetConnectionException(  "TCP " + port + " is not defined on your Application Port Mapping section");
+            } 
         }
 
-
-        public async Task<String> GetURI()
+        /// <summary>
+        /// Returns the URI for your backend (server) deployed on MobiledgeX
+        /// </summary>
+        /// <param name="protocol">(LProto) Protocol (TCP, UDP)</param>
+        /// <param name="port">(Integer) Desired Port </param>
+        /// <returns></returns>
+        public async Task<String> GetURI(LProto protocol = LProto.L_PROTO_TCP, int port = 0 )
         {
-            ConfigureMobiledgeXSettings();
-            string host="";
-            string port="";
-            // For Demo App purposes, it's the TCP app port. Your app may point somewhere else:
-            NetTest.Site site;      
-            string aCarrierName = GetCarrierName();
-            Debug.Log("aCarrierName: " + aCarrierName);
+            await ConfigureMobiledgeXSettings();
+            string uri = "";
+            string host = "";
+            if (port == 0)
+            {
+                port = tcpPort;
+            }
+            NetTest.Site site;
             Debug.Log("Calling DME to register client...");
             bool registered = false;
             registered = await Register();
 
             if (!registered)
             {
-                Debug.Log("Exceptions, or app not found. Not Registered!");
-                return null;
+                Debug.LogError("MobiledgeX: Make sure your credentials in MobiledgeX Settings are the same as on MobiledgeX Console.\n check the exceptions for more info.");
+                throw new RegisterClientException("RegisterClient failed");
             }
             else
             {
                 FindCloudletReply reply;
                 Debug.Log("Finding Cloudlet...");
                 reply = await FindCloudlet();
 
```

Comment:
I don't think you should reimplement this. We already have FindCloudlet with Performance mode in C# SDK